utils/dom.js

```diff
@@ -58,7 +58,12 @@ export function isHorizontalEdge( container, isReverse, collapseRanges = false )
 
 	const maxOffset = node.nodeType === TEXT_NODE ? node.nodeValue.length : node.childNodes.length;
 
-	if ( ! isReverse && offset !== maxOffset ) {
+	if (
+		! isReverse &&
+		offset !== maxOffset &&
+		// content editables with only a BR element are considered empty
+		( offset !== 0 || maxOffset !== 1 || node.nodeType === TEXT_NODE || ! node.childNodes.length || node.childNodes[ 0 ].tagName !== 'BR' )
```

Comment:
I find this whole thing quite hard to follow. Would it be possible the separate this case? Also, this seems to look for `BR` deeply nested too, while it seems we might just want the direct child of the container?

Reply:
I wonder if we can add an `isEmpty` helper, check for that with the `container` and return true early.

Reply:
Hm, I wonder why this worded previously... https://github.com/WordPress/gutenberg/pull/5159/files
No br checks .

Reply:
Most likely because MCE corrects wrong ranges.

Reply:
Updated to rely on TinyMCE API instead.